src/Features/Core/Portable/NavigateTo/NavigateToSearcher.cs

```diff
@@ -221,54 +221,79 @@ private ImmutableArray<ImmutableArray<Project>> GetOrderedProjectsToProcess()
         {
             using var result = TemporaryArray<ImmutableArray<Project>>.Empty;
 
-            using var _ = PooledHashSet<Project>.GetInstance(out var processedProjects);
-
-            // First, if there's an active document, search that project first, prioritizing that active document and
-            // all visible documents from it.
-            if (_activeDocument != null)
+            // Get the initial set of project groups.  But filter out any projects that don't have an associated search
+            // service.  No point examining them or adding progress items for them.
+            foreach (var group in GetOrderedProjectsToProcessWorker())
             {
-                processedProjects.Add(_activeDocument.Project);
-                result.Add(ImmutableArray.Create(_activeDocument.Project));
-            }
+                using var groupCopy = TemporaryArray<Project>.Empty;
```

Comment:
>groupCopy [](http://example.com/codeflow?start=26&length=9)

Isn't the expected number of projects in this array going to be greater than 4? In such case TemporaryArray is not very efficient.